app/src/main/java/org/jellyfin/mobile/player/PlayerViewModel.kt

```diff
@@ -168,6 +182,30 @@ class PlayerViewModel(application: Application) : AndroidViewModel(application),
                 else -> DefaultRenderersFactory.EXTENSION_RENDERER_MODE_ON
             }
             setExtensionRendererMode(rendererMode)
+            // set the media selector
+            setMediaCodecSelector { mimeType, requiresSecureDecoder, requiresTunnelingDecoder ->
+                val decoderInfoList = MediaCodecSelector.DEFAULT.getDecoderInfos(mimeType, requiresSecureDecoder, requiresTunnelingDecoder)
+                // allow decoder selection only for video
+                if (!MimeTypes.isVideo(mimeType)) {
+                    return@setMediaCodecSelector decoderInfoList
+                }
+                val filteredDecoderList = when (decoderType.value) {
+                    DecoderType.HARDWARE -> {
+                        // only get the hardware decoder
+                        decoderInfoList.filter { it.hardwareAccelerated }
+                    }
+                    DecoderType.SOFTWARE -> {
+                        decoderInfoList.filter { !it.hardwareAccelerated }
+                    }
```

Comment:
```suggestion
                    DecoderType.HARDWARE -> decoderInfoList.filter(MediaCodecInfo::hardwareAccelerated)
                    DecoderType.SOFTWARE -> decoderInfoList.filterNot(MediaCodecInfo::hardwareAccelerated)
```

Reply:
Maybe also remove the comment and curly braces here, like I suggested. I don't think an extra comment is necessary in this case.